docs/tsp/typeServerProtocol.ts

```diff
@@ -241,36 +269,35 @@ export namespace TypeServerProtocol {
         // First request to initialize the type server.
         Initialize = 'typeServer/initialize',
         // Request from client to shut down the type server.
-        Shutdown = 'typeServer/shutdown',
+        ShutDown = 'typeServer/shutdown',
         // Request from client to get the current snapshot of the type server.
         // A snapshot is a point-in-time representation of the type server's state, including all loaded files and their types.
         // A type server should change its snapshot whenever any type it might have returned is no longer valid. Meaning types are
         // only usable for the snapshot they were returned with.
+        //
+        // Snapshots are not meant to survive any changes that would make the type server throw away its internal cache. They are merely an
+        // identifier to indicate to the client that the type server will accept requests for types from that snapshot.
         GetSnapshot = 'typeServer/getSnapshot',
         // Request to get diagnostics for a specific file.
         GetDiagnostics = 'typeServer/getDiagnostics',
```

Comment:
Similar question: what if the server wants to push new diagnostics to the client? We might see file watch events that VSCode's file watcher does not.

Reply:
We could handle this with another notification, diagnostics changed or something along those lines.